src/NBitcoin/MutableConsensus.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using NBitcoin.BouncyCastle.Math;
+
+namespace NBitcoin
+{
+    /// <summary>
+    /// Used to create a Mutable version of the <see cref="Consensus"/> object during configuration
+    /// </summary>
+    public class MutableConsensus : IConsensus
```

Comment:
I am not sure the naming is correct but I like it none the less.

Reply:
I wonder if instead of using an interface we just use an Abstract consensus class with protected setters this will make it immutable and not need to have duplicated properties (on interface and two implementations)

Reply:
I think it is better to keep an interface that we can easily mock when doing tests